cfme/tests/cloud_infra_common/test_custom_attributes_rest.py

```diff
@@ -56,9 +56,13 @@ def get_vm(appliance, provider, vm_obj):
         collection = appliance.rest_api.collections.instances
 
     def _get_vm():
+        if not provider.mgmt.does_vm_exist(vm_obj.name):
+            vms = collection.find_by(name=vm_obj.name)
```

Comment:
QUESTION: why is this necessary? If we are in this branch, the VM doesn't exist so ``find_by`` will not return anything, right?

Reply:
collection checks for VM in CFME. this test being parallelized and module-scoped - is executed simultaneously with the same `vm_name` on different appliances. which would lead to `create_on_provider` if vm not found on the appliance

Reply:
@lkhomenk Yes, but I don't see a point in calling ``collection.find_by(..)`` when we already checked that the VM doesn't exist on provider. Could this part be just:
```python
if not provider.mgmt.does_vm_exist(vm_obj.name):
    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip='default')
```
?

Reply:
@mkoura if VM exists on the provider but missing in CFME - we'll fail getting VM's REST representation - `vms = collection.find_by(name=vm_obj.name)`

Reply:
@lkhomenk that's my point. If the vm doesn't exist on provider, there's no point checking if it exists in CFME - it will always fail. Checking if it exists on provider and creating it if doesn't is enough.